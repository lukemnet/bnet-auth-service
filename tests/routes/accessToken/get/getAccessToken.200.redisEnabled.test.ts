import fastify from 'fastify';
import fastifyRedis from 'fastify-redis-mock';
import server from '../../../../src/index';
import getConfig from '../../../helper';

describe('/accessToken/get 200 (Redis enabled)', () => {
  const fastifyServer = fastify() as any;

  beforeAll(async () => {
    fastifyServer.register(fastifyRedis, {
      host: '127.0.0.1',
      port: '6379',
      password: '',
      enableReadyCheck: true,
      dropBufferSupport: false,
    });
    await fastifyServer.register(server, getConfig(true));
  });

  afterAll(() => {
    fastifyServer.close();
  });

  it('returns 200', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(JSON.parse(res.payload)).toEqual({"status":200,"data":{ accessToken: "sample access token" }});
  });

  it('returns correct response when access token is cached', async () => {
    await (fastifyServer as any).redis.set(getConfig(true).redis.cacheSegment, 'sample cached access token');
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(JSON.parse(res.payload)).toEqual({"status":200,"data":{ accessToken: "sample cached access token" }});
  });

  it('returns 200 when refresh parameter is sent', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get?refresh=true', });
    expect(res.statusCode).toBe(200);
  });

  it('returns correct response refresh parameter is sent', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get?refresh=true', });
    expect(JSON.parse(res.payload)).toEqual({"status":200,"data":{ accessToken: "sample access token" }});
  });

  it('response is cached correctly', async () => {
    await fastifyServer.inject({ method: 'GET', url: '/accessToken/get?refresh=true', });
    const cachedResponse = await fastifyServer.redis.get(getConfig(true).redis.cacheSegment);
    expect(cachedResponse).toEqual("sample access token");
  });
});