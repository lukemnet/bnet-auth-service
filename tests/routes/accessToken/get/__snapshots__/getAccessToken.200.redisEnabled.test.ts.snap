// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/accessToken/get 200 (Redis enabled) returns correct response 1`] = `"{\\"status\\":200,\\"data\\":{\\"accessToken\\":\\"sample access token\\"}}"`;

exports[`/accessToken/get 200 (Redis enabled) returns correct response refresh parameter is sent 1`] = `"{\\"status\\":200,\\"data\\":{\\"accessToken\\":\\"sample access token\\"}}"`;

exports[`/accessToken/get 200 (Redis enabled) returns correct response when access token is cached 1`] = `"{\\"status\\":200,\\"data\\":{\\"accessToken\\":\\"sample cached access token\\"}}"`;
